### PR TITLE
Implement missing orbit velocity for CPUParticles and CPUParticles2D

### LINKED ISSUE #29580: Missing FLAG_DISABLE_Z enum value in CPU Particles(2D)
**Godot version:**
3.2.dev.custom_build.197b65f32

**Issue description:**
Similar issue as #29419, but this time in CPU Particles( maybe also CPU Particles 2D, but I'm not sure)
Zrzut ekranu z 2019-06-07 17-41-34
Zrzut ekranu z 2019-06-07 17-41-38

also in CPU Particles 3D can be missing PARAM_ORBIT_VELOCITY which is present in CPU Particles 2D(Not sure, but in code is commented)

### PR DESCRIPTION
**Pull request by akien-mga**
The relevant code was copied from (GPU) ParticlesMaterial but commented
out initially, and never ported.

~The ported code seems to work in a quick test, but I don't manage to get
Particles2D and CPUParticles2D to behave the same for the same parameters,
so it's hard to assess if it's done right.~

~Once I'm confident that it works as expected, I'll port it to CPUParticles (3D).~
Done.

Closes #29580.

Part of #29692.

### CODE REVIEW

**Comment by clayjohn** [comment]
Looks good to me so far. There should be no real challenge to implementing it. Would be nice to do at the same time as the rest of https://github.com/godotengine/godot/issues/29692

**Comment by akien-mga** [comment]
Should be ready for review now.

I solved the direction issue I had by creating the rotation matrix (`Transform2D`) with `-ang` instead of `ang`. The matching `ParticlesMaterial` code seems indeed to use a clockwise rotation matrix:
https://github.com/godotengine/godot/blob/master/scene/resources/particles_material.cpp#L447-L448

See https://en.wikipedia.org/wiki/Rotation_matrix#In_two_dimensions for comparison.
I can't say if this was done on purpose, or if it was a mistake as the matrix is constructed with `mat2(vec2, vec2)`, each `vec2` being a column, but if they were lines it would be the counterclockwise rotation matrix.

I also dropped `FLAG_DISABLE_Z` from the `CPUParticles2D` code as this flag seems to only serve to distinguish between 2D and 3D in the common `ParticlesMaterial`. I kept it in `CPUParticles` though, as in `Particles` you can set Disable Z to emulate 2D behavior in 3D.

File: scene/2d/cpu_particles_2d.cpp [hunk 1]
```diff
@@ -759,18 +750,15 @@ void CPUParticles2D::_particles_process(float p_delta) {
...
-					float ang = orbit_amount * DELTA * pi * 2.0;
-					mat2 rot = mat2(vec2(cos(ang), -sin(ang)), vec2(sin(ang), cos(ang)));
-					TRANSFORM[3].xy -= diff.xy;
-					TRANSFORM[3].xy += rot * diff.xy;
-				}
+			float orbit_amount = (parameters[PARAM_ORBIT_VELOCITY] + tex_orbit_velocity) * Math::lerp(1.0f, rand_from_seed(alt_seed), randomness[PARAM_ORBIT_VELOCITY]);
+			if (orbit_amount != 0.0) {
+				float ang = orbit_amount * local_delta * Math_PI * 2.0;
+				// Not sure why the ParticlesMaterial code uses a clockwise rotation matrix,
+				// but we use -ang here to reproduce its behavior.
```
Comment (akien-mga) [inline scene/2d/cpu_particles_2d.cpp]: Anyway, that's how the pro graphics engineer do maths, no? Tweak signs and test until it works. :thinking:

Comment (JFonS) [reply scene/2d/cpu_particles_2d.cpp]: Yes, that's exactly how graphics engineers do maths. It falls into a wider technique called "permutation programming" in which you try all the possible permutations of operations (matrix multiplications, negation signs, etc.) until you get the desired result.

**Review by JFonS — APPROVED** [review]
Looks good to me.

**Pull request merged**